integrations/jira/jira/client.py

```diff
@@ -103,107 +180,75 @@ async def create_events_webhook(self, app_host: str) -> None:
             "events": WEBHOOK_EVENTS,
         }
 
-        webhook_create_response = await self.client.post(
-            f"{self.webhooks_url}", json=body
-        )
-        webhook_create_response.raise_for_status()
+        await self._send_api_request("POST", self.webhooks_url, json=body)
         logger.info("Ocean real time reporting webhook created")
 
-    async def get_single_project(self, project_key: str) -> dict[str, Any]:
-        project_response = await self.client.get(
-            f"{self.api_url}/project/{project_key}"
+    async def get_single_project(self, project_key: str) -> Dict[str, Any]:
+        return await self._send_api_request(
+            "GET", f"{self.api_url}/project/{project_key}"
         )
-        project_response.raise_for_status()
-        return project_response.json()
 
     async def get_paginated_projects(
         self,
-    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+    ) -> AsyncGenerator[List[Dict[str, Any]], None]:
```

Comment:
Let's favour the built-in types for type hints over types from the `typing` module, it is fully supported by this Python version over the older `typing.<type>` version.